pkg/recorder/recorder.go

```diff
@@ -0,0 +1,475 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package recorder
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/maps/recorder"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/u8proto"
+)
+
+const (
+	subsystem = "recorder"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, subsystem)
+
+type ID uint16
+
+type RecorderTuple struct {
+	SrcPrefix net.IPNet
+	SrcPort   uint16
+	DstPrefix net.IPNet
+	DstPort   uint16
+	Proto     u8proto.U8proto
+}
+
+type RecorderMask struct {
+	srcMask net.IPMask
+	srcPort uint16
+	dstMask net.IPMask
+	dstPort uint16
+	proto   u8proto.U8proto
+}
+
+type RecInfo struct {
+	ID      ID
+	CapLen  uint16
+	Filters []RecorderTuple
+}
+
+type RecMask struct {
+	users int
+	prio  int
+	mask  RecorderMask
+}
+
+type RecQueue struct {
```

Comment:
nit: This type should be made private
```suggestion
type recQueue struct {
```